external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -73,6 +74,11 @@ private[kafka010] class KafkaMicroBatchStream(
 
   private var endPartitionOffsets: KafkaSourceOffset = _
 
+  private lazy val offsetMetrics = new KafkaCommitsSource()
+
+  private val shouldSetCommitOffsetsOnCheckpoints =
+    Option(options.get("setCommitOffsetsOnCheckpoints")).exists(_.toBoolean)
```

Comment:
Hi @skonto . 
Could you add some test coverage for this new option, please?

Reply:
Sure I will do.